File: Java/Java source-code/errorRate.java


public class errorRate {
    public static double errorRate_m(Double[] labels, Double[] predictions) {
        double nberrors = 0;
        double datasize = 0;
        for (int index = 0; index < labels.length; index++) {
            if (predictions[index] == -1) {
                continue;
            }
            if (predictions[index] != labels[index]) {
                nberrors++;
            }
            datasize++;
        }
        return nberrors / datasize;
    }
}

File: Methods/Methods-Python/errorRate.py
def errorRate(labels, predictions):
    nberrors = 0
    datasize = 0

    for i in range(0, len(labels)):
        if predictions[i] == -1:
            continue

        if predictions[i] != labels[i]:
            nberrors += 1

        datasize += 1

    return nberrors / datasize
